examples/mixed_signal/tests/test_regulator.py

```diff
@@ -0,0 +1,242 @@
+import cocotb
+from cocotb.regression import TestFactory
+from cocotb.triggers import Timer
+from cocotb.utils import get_sim_time
+
+from collections import namedtuple
+
+import datetime
+import itertools
+import logging
+import matplotlib
+import matplotlib.pyplot as plt
+
+
+Dataset = namedtuple('Dataset', 'time, trim, voltage, current')
+
+class Regulator_TB(object):
+    """Class for collecting testbench objects.
+
+    Args:
+        dut: The toplevel of the design-under-test.
+            In this mixed cocotb/HDL testbench environment, it is the HDL testbench.
+        settling_time_ns (int): Time in nanoseconds to wait before sample is taken.
+    """
+
+    def __init__(self, dut, settling_time_ns=1):
+        self.dut = dut
+        self.settling_time_ns = settling_time_ns
+        self.analog_probe = dut.i_analog_probe  #: The instance name of the analog probe module.
+        self._bittogglestream = itertools.cycle(range(2))  #: Produce stream that toggles between ``0`` and ``1``.
```

Comment:
nit: `[0, 1]` is a short way of spelling `range(2)` that as a bonus matches your comment